nixos/modules/services/torrent/transmission.nix

```diff
@@ -1,52 +1,52 @@
-
-  homeDir = cfg.home;
-  downloadDirPermissions = cfg.downloadDirPermissions;
-  downloadDir = "${homeDir}/Downloads";
-  incompleteDir = "${homeDir}/.incomplete";
-
-  settingsDir = "${homeDir}/config";
-  settingsFile = pkgs.writeText "settings.json" (builtins.toJSON fullSettings);
-
-  # for users in group "transmission" to have access to torrents
-  fullSettings = { umask = 2; download-dir = downloadDir; incomplete-dir = incompleteDir; } // cfg.settings;
-
-  preStart = pkgs.writeScript "transmission-pre-start" ''
-    #!${pkgs.runtimeShell}
-    set -ex
-    cp -f ${settingsFile} ${settingsDir}/settings.json
-  '';
+  stateDir = "/var/lib/transmission";
+  # TODO: switch to configGen.json once RFC0042 is implemented
+  settingsFile = pkgs.writeText "settings.json" (builtins.toJSON (cfg.settings // {
+    download-dir = "${stateDir}/Downloads";
+    incomplete-dir = "${stateDir}/.incomplete";
+  }));
+  settingsDir = ".config/transmission-daemon";
+  makeAbsolute = base: path:
+    if builtins.match "^/.*" path == null
+    then base+"/"+path else path;
 in
 {
   options = {
     services.transmission = {
-      enable = mkOption {
-        type = types.bool;
-        default = false;
-        description = ''
-          Whether or not to enable the headless Transmission BitTorrent daemon.
+      enable = mkEnableOption ''
+        Whether or not to enable the headless Transmission BitTorrent daemon.
 
-          Transmission daemon can be controlled via the RPC interface using
-          transmission-remote or the WebUI (http://localhost:9091/ by default).
+        Transmission daemon can be controlled via the RPC interface using
+        transmission-remote, the WebUI (http://${cfg.settings.rpc-bind-address}:${toString cfg.settings.rpc-port}/ by default),
+        or other clients like stig or tremc.
 
-          Torrents are downloaded to ${downloadDir} by default and are
-          accessible to users in the "transmission" group.
-        '';
-      };
+        Torrents are downloaded to ${cfg.settings.download-dir} by default and are
+        accessible to users in the "transmission" group.
+      '';
 
-      settings = mkOption {
+      settings = mkOption rec {
+        # TODO: switch to types.config.json as prescribed by RFC0042 once it's implemented
         type = types.attrs;
+        apply = attrs:
+          let super = recursiveUpdate default attrs; in
+          super // {
+            download-dir   = makeAbsolute cfg.home super.download-dir;
+            incomplete-dir = makeAbsolute cfg.home super.incomplete-dir;
+          };
         default =
           {
-            download-dir = downloadDir;
-            incomplete-dir = incompleteDir;
+            download-dir = "${cfg.home}/Downloads";
+            incomplete-dir = "${cfg.home}/.incomplete";
             incomplete-dir-enabled = true;
+            rpc-bind-address = "127.0.0.1";
+            rpc-port = 9091;
+            umask = 63; # echo $((8#077))
```

Comment:
What is this echo comment?

Reply:
A command to convert octal to decimal as expected by Transmission due to the JSON encoding, I've improved the comment.